python/cudf/dataframe/multiindex.py

```diff
@@ -85,7 +90,7 @@ def _validate_levels_and_codes(self, levels, codes):
                                  'than maximum level size at this position')
 
     def copy(self, deep=True):
-        mi = MultiIndex(self.levels.copy(),
+        mi = MultiIndex(self.levels,
```

Comment:
This will materialize the levels and codes, you want to pass `self._levels` and `self._codes` here.

Reply:
I rewrote copy and wrote tests for it as well.